src/dmd/clone.d

```diff
@@ -570,18 +570,15 @@ FuncDeclaration buildXopEquals(StructDeclaration sd, Scope* sc)
     Loc declLoc; // loc is unnecessary so __xopEquals is never called directly
     Loc loc; // loc is unnecessary so errors are gagged
     auto parameters = new Parameters();
-    parameters.push(new Parameter(STC.ref_ | STC.const_, sd.type, Id.p, null, null))
-              .push(new Parameter(STC.ref_ | STC.const_, sd.type, Id.q, null, null));
-    auto tf = new TypeFunction(ParameterList(parameters), Type.tbool, LINK.d);
+    parameters.push(new Parameter(STC.ref_ | STC.const_, sd.type, Id.p, null, null));
+    auto tf = new TypeFunction(ParameterList(parameters), Type.tbool, LINK.d, STC.const_);
     Identifier id = Id.xopEquals;
-    auto fop = new FuncDeclaration(declLoc, Loc.initial, id, STC.static_, tf);
+    auto fop = new FuncDeclaration(declLoc, Loc.initial, id, 0, tf);
     fop.generated = true;
-    Expression e1 = new IdentifierExp(loc, Id.p);
-    Expression e2 = new IdentifierExp(loc, Id.q);
-    // TODO: simplify as soon as `git describe` for DMD master yields v2.099+
-    Expression e = global.versionNumber() >= 2099
-        ? new EqualExp(EXP.equal, loc, e2, e1)
-        : new EqualExp(EXP.equal, loc, e1, e2);
+    fop.parent = sd;
+    Expression e1 = new IdentifierExp(loc, Id.This);
+    Expression e2 = new IdentifierExp(loc, Id.p);
+    Expression e = new EqualExp(EXP.equal, loc, e1, e2);
```

Comment:
With `global.versionNumber() < 2099`, druntime will still call it in reversed order (`this` here being `p2`), so you'll need to keep the previous ternary expression with inverted condition.

Reply:
That... Doesn't matter? Because this is targeting a 2.099 library.

Reply:
? What else would have caused the CI regressions?

Reply:
> ? What else would have caused the CI regressions?

That is still yet to be determined. What I saw was unexpected compiler errors, druntime doesn't kick in just yet.

Reply:
Ah-ha, apparently `STC.const_` is not enough anymore when constructing a `TypeFunction`, you also need to call `addSTC(STC.const_)` too.

Let's see what the CI now says...

Reply:
> Ah-ha, apparently STC.const_ is not enough anymore when constructing a TypeFunction

Ouch, apparently `STC.const_` is totally ignored by the `TypeFunction` ctor.

---

I think my point above still holds; this should now make a wrapped `opEquals` by-value method invoked in reversed order via `TypeInfo_Struct.equals()` for as long as `__VERSION__` is still at `2098`.

Reply:
But `__VERSION__` is now 2099, and again, this is the code generator, not the runtime. So this part of the frontend is unaffected by the swap.

Saying that, it just occurred to me that you might be talking about the master->stable merge - but no - if druntime is merged first then it still doesn't affect this code.

Reply:
> But __VERSION__ is now 2099

What? Not if you have a `.git` subdirectory in the DMD repo root; the checked-in `VERSION` file is ignored in that case. It's what made everything a bit more complex for the other PR.

Reply:
Hmm, all the more curious then that this is succeeding.